server/src/main/java/de/uftos/repositories/ucdl/parser/DefinitionParser.java

```diff
@@ -0,0 +1,691 @@
+                  return elementInSet;
+                }
+                params = new ArrayList<>();
+                params.add(elementInSet);
+                params.add(buildAst(root.jjtGetChild(2).jjtGetChild(0), parameters));
+                return switch (((SimpleNode) root.jjtGetChild(2)).jjtGetFirstToken().image) {
+                  case "implies", "->" -> new OperatorDto(UcdlToken.IMPLIES, params);
+                  case "or", "||" -> new OperatorDto(UcdlToken.OR, params);
+                  case "and", "&&" -> new OperatorDto(UcdlToken.AND, params);
+                  default -> throw new IllegalStateException();
+                };
+              case "EQUATION":
+                params = new ArrayList<>();
+                Node firstElement = root.jjtGetChild(0);
+                Node secondElement = root.jjtGetChild(1).jjtGetChild(1);
+                String comparator =
+                    ((SimpleNode) root.jjtGetChild(1).jjtGetChild(0)).jjtGetFirstToken().image;
+
+                params.add(buildAst(firstElement, parameters));
+                params.add(buildAst(secondElement, parameters));
+                UcdlToken token = getComparatorToken(comparator, params);
+                return new OperatorDto(token, params);
+              default:
+                throw new IllegalStateException();
+            }
+          default:
+            throw new IllegalStateException();
+
+        }
+      }
+      //includes cases: "CONTROLSEQUENCELIST", "CONTROLSEQUENCERETURN",
+      // "OPTIONALIMPLIES", "OPTIONALOR", "OPTIONALAND",
+      // "ELEMENTINSET", "EQUATION", "ELEMENTEQUATION", "ELEMENTATTRIBUTELIST",
+      // "SETNAME", "SETMODIFICATION", "NUMBERLIST", "ATTRIBUTE", "FILTERLIST"
+      default -> throw new IllegalStateException();
+    }
+  }
+
+  //checks that all returnValues are the same and returns the value
+  // (throws an exception if true and false are returnValues inside the body)
+  private static boolean getReturnValue(List<AbstractSyntaxTreeDto> body) throws ParseException {
+    boolean and = true;
+    boolean or = false;
+    for (AbstractSyntaxTreeDto ast : body) {
+      if (ast.getToken() == UcdlToken.FOR || ast.getToken() == UcdlToken.IF) {
+        ControlSequenceDto cs = (ControlSequenceDto) ast;
+        and = and && cs.returnValue();
+        or = or || cs.returnValue();
+      }
+    }
+    if (and != or) {
+      throw new ParseException(
+          "All control sequences in the body of a control sequence need to return"
+              + " the same boolean value!");
+    }
+    return and;
+  }
+
+  private static UcdlToken getComparatorToken(String comparator, List<AbstractSyntaxTreeDto> params)
+      throws ParseException {
+    UcdlToken token = switch (comparator) {
+      case ">" -> UcdlToken.GREATER;
+      case "<" -> UcdlToken.SMALLER;
+      case ">=" -> UcdlToken.GREATER_EQUALS;
+      case "<=" -> UcdlToken.SMALLER_EQUALS;
+      case "=", "==" -> UcdlToken.EQUALS;
```

Comment:
Shouldn't we reserve `=` in case ucdl should ever get assignments? Also, a single equal sign is just confusing to me 😅

Reply:
I think it was left in for users who might not be familiar with programming.
But we can remove it (however this will require generating a new parser).